### ISSUE TITLE
Investigate input purposes for address pattern

### ISSUE THREAD

**Issue by nickcolley** [alphagov/govuk-design-system#1090]
When we added autocomplete values to our examples we skipped addresses as it's unclear the appropriate values and we did not want to block shipping the overall work.

> An outstanding change which is not part of this PR is to update the [multiple text inputs example on addresses](https://design-system.service.gov.uk/patterns/addresses/#multiple-text-inputs) to set the autocomplete attribute on each input. This is because we are unsure of how the [input purposes](https://www.w3.org/TR/WCAG21/#input-purposes) for address fields should be correctly used. We are investigating this, but decided not to let it block the PR. 

https://github.com/alphagov/govuk-design-system/pull/727#issuecomment-460347140

It would be good to investigate this now and see if there's any improvements we can make to the pattern, arguably this is a WCAG 2.1 compliance requirement as input purposes are needed where appropriate.

**Comment by colinrotherham** [comment]
The spec is US-centric and quite confusing for UK addresses.

For the "Multiple text inputs" address pattern:
https://design-system.service.gov.uk/patterns/addresses/#multiple-text-inputs

Our service went with:

**Lines 1 and 2**
```
address-line1
address-line2
```

**Town or city**
```
address-level2
```

**County**
```
address-level1
```

**Postcode**
```
postal-code
```

Why? Because logically the levels go in reverse from broad to fine grained:

_address-level1_ (province/state)
_address-level2_ (city/town)
_address-level3_ (for addresses with 3 administrative levels)
_address-level4_ (for addresses with 4 administrative levels) 

### Research

From the Google Chrome intro to autofill:
https://developers.google.com/web/updates/2015/06/checkout-faster-with-autofill

> For one address input: street-address For two address inputs: address-line1 , address-line2 address-level1 (state or province) address-level2 (city) postal-code (zip code) country

From MDN
https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete

> **“address-level1”**
> The first administrative level in the address. This is typically the province in which the address is located. In the United States, this would be the state. In Switzerland, the canton. In the United Kingdom, the post town.

> **“address-level2"**
> The second administrative level, in addresses with at least two of them. In countries with two administrative levels, this would typically be the city, town, village, or other locality in which the address is located.

From the WHATWG spec:
https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#attr-fe-autocomplete-address-level1

> **“address-level1”**
> The broadest administrative level in the address, i.e. the province within which the locality is found; for example, in the US, this would be the state; in Switzerland it would be the canton; in the UK, the post town

> **“address-level2"**
> The second administrative level, in addresses with two or more administrative levels; in the countries with two administrative levels, this would typically be the city, town, village, or other locality within which the relevant street address is found

### In summary
But… further reading of the WHATWG spec is needed.

The spec implies that in the UK we have no autocomplete attribute for "County" because it's not necessary in a UK address.

This is backed up by [comments from Ian Hickson](https://readable-email.org/list/whatwg/topic/supporting-more-address-levels-in-autocomplete#content-3).

>(Note that a lot of people in the UK have no idea how to write their address according to current standards. For example, people often include the county, give the "real" town rather than the "post town", put things out of order, indent each line of the address, etc.)

Do we keep using **“address-level1”** for County based on granularity? This complies with spec comments "The "address-level1" field represents the widest administrative division".

**Further reading**
Looks like "County" was dropped from UK addresses in 2010:
https://www.bbc.co.uk/news/magazine-10853705

**Comment by terrysimpson99** [comment]
The "County" question is also at: https://github.com/alphagov/govuk-design-system-backlog/issues/31
